### DISCUSSION TITLE
jsonLogic support for array operations like "some"

### DISCUSSION THREAD

**Discussion by nicopadu** [react-querybuilder/react-querybuilder#634] [Q&A] [answered]
Given the structure of our source data, we'd need to deal with Array operators from jsonLogic.
Specifically with "some" operator, combined with "in".
https://jsonlogic.com/operations.html#all-none-and-some

Checking in the code, there is some commented code related to unused operations in this file
https://github.com/react-querybuilder/react-querybuilder/blob/main/packages/react-querybuilder/src/utils/parseJsonLogic/utils.ts

Are they not supported due to any known issue? Or just that this requirement was not relevant or present at the moment of jsonLogic support implementation?

I guess that supporting these kind of operators would require also extra settings in Fields properties?
Something like "group" type of fields as here https://github.com/ukrbublik/react-awesome-query-builder/blob/master/CONFIG.adoc#configfields

**Comment by jakeboone02** [comment, marked as answer]
Your guess about this not being present at the moment of JsonLogic support implementation is correct.

I'm pretty sure we could do this, with some caveats. One of those caveats would be that the reverse operation (`formatQuery(query, 'jsonlogic')`) would almost certainly not yield the same "some"/"all"/"none" rule that `parseJsonLogic` would have parsed, at least by default. The workaround would be to use a [custom rule parser](https://react-querybuilder.js.org/docs/utils/import#custom-operations) and probably a [custom rule formatter](https://react-querybuilder.js.org/docs/utils/export#rule-processor) as well.

Let me see what I can come up with. I'll create an issue from this discussion and track it there (edit: see #635).
